spring-integration-file/src/test/java/org/springframework/integration/file/tail/FileTailingMessageProducerTests.java

```diff
@@ -124,9 +127,47 @@ public void canRecalculateCommandWhenFileOrOptionsChanged() throws IOException {
 		adapter.stop();
 	}
 
+	@Test
+	@TailAvailable
+	public void testIdleEvent() throws Exception {
+		OSDelegatingFileTailingMessageProducer adapter = new OSDelegatingFileTailingMessageProducer();
+		adapter.setOptions(TAIL_OPTIONS_FOLLOW_NAME_ALL_LINES);
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.afterPropertiesSet();
+		adapter.setTaskScheduler(taskScheduler);
+		final List<FileTailingIdleEvent> events = new ArrayList<>();
+		adapter.setApplicationEventPublisher(event -> {
+			if (event instanceof FileTailingIdleEvent) {
+				FileTailingIdleEvent tailEvent = (FileTailingIdleEvent) event;
+				logger.warn(event);
+				events.add(tailEvent);
+			}
+		});
+		File file = new File(testDir, "foo");
+		file.delete();
+		FileOutputStream foo = new FileOutputStream(file);
+		for (int i = 0; i < 50; i++) {
+			foo.write(("hello" + i + "\n").getBytes());
+		}
+		foo.flush();
+		foo.close();
+		adapter.setFile(file);
+		QueueChannel outputChannel = new QueueChannel();
+		adapter.setOutputChannel(outputChannel);
+		adapter.setIdleEventInterval(1000);
```

Comment:
Since we would like to be sure that we emit `FileTailingIdleEvent` anyway, that would be better to have `idleEventInterval` as small as possible to make the test case reasonably quick.